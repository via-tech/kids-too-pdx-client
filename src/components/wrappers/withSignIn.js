import React from 'react';
import PropTypes from 'prop-types';

export const withSignIn = Component => {
  function WithSignIn(props) {
    const { token, history } = props;

    if(!token) return history.push('/login');

    return <Component {...props} />;
  }

  WithSignIn.propTypes = {
    token: PropTypes.string,
    history: PropTypes.object
  };

  return WithSignIn;
};
