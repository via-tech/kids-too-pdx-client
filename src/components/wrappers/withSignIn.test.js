import React from 'react';
import renderer from 'react-test-renderer';
import { withSignIn } from './withSignIn';

describe('With sign in wrapper', () => {
  const Component = withSignIn(() =>
    <div>
      <h1>The Main Component</h1>
    </div>
  );

  it('snapshots withSignIn signed out', () => {
    const props = {
      token: '',
      history: {
        push(route) {
          return route;
        }
      }
    };

    const tree = renderer.create(
      <Component {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots withSignIn signed in', () => {
    const props = {
      token: '1234',
      history: {
        push(route) {
          return route;
        }
      }
    };

    const tree = renderer.create(
      <Component {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
