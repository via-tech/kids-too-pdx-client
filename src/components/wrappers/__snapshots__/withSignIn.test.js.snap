// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With sign in wrapper snapshots withSignIn signed in 1`] = `
<div>
  <h1>
    The Main Component
  </h1>
</div>
`;

exports[`With sign in wrapper snapshots withSignIn signed out 1`] = `"/login"`;
